### PR TITLE
Deafult set cluster status as unhealthy if deleted due to TTL

### PR DESCRIPTION
**Pull request by shtripat**
Signed-off-by: Shubhendu <shtripat@redhat.com>

### CODE REVIEW

File: tendrl/integrations/gluster/check_cluster_status.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import etcd
+
+
+def run():
+    try:
+        NS._int.client.write(
+            "/clusters/{0}/GlobalDetails/status".format(
+                NS.tendrl_context.integration_id
```
Comment (nnDarshan) [inline tendrl/integrations/gluster/check_cluster_status.py]: This is monitor node, I guess it wont have integration id. I feel you have to loop through all the clusters and set the status

Comment (shtripat) [reply tendrl/integrations/gluster/check_cluster_status.py]: Ack. Will do that.

**Comment by shtripat** [comment]
Verified as per https://github.com/Tendrl/gluster-integration/pull/526#issuecomment-348681186

File: tendrl/integrations/gluster/check_cluster_status.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import etcd
+
+
+def run():
+    try:
+        clusters = NS._int.client.read("/clusters")
```
Comment (r0h4n) [inline tendrl/integrations/gluster/check_cluster_status.py]: use commons.utils.etcd_util.read

Comment (r0h4n) [inline tendrl/integrations/gluster/check_cluster_status.py]: add a comment that this is only going to be run by the "tendrl/monitor'

File: tendrl/integrations/gluster/check_cluster_status.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    except etcd.EtcdKeyNotFound:
+        return
+
+    for cluster in clusters.leaves:
+        int_id = cluster.key.split('/')[-1]
+        try:
+            NS._int.client.write(
```
Comment (r0h4n) [inline tendrl/integrations/gluster/check_cluster_status.py]: do this only for is_managed == "yes" clusters

File: tendrl/integrations/gluster/check_cluster_status.py (hunk 2 above)
Comment (shtripat) [reply tendrl/integrations/gluster/check_cluster_status.py]: Done

Comment (shtripat) [reply tendrl/integrations/gluster/check_cluster_status.py]: Sure

File: tendrl/integrations/gluster/check_cluster_status.py (hunk 3 above)
Comment (shtripat) [reply tendrl/integrations/gluster/check_cluster_status.py]: Ah ok. will change the logic and update

File: tendrl/integrations/gluster/check_cluster_status.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    except etcd.EtcdKeyNotFound:
+        return
+
+    # This logic only runs on node with tag `tendrl/monitor` (tendrl server)
+    # So its done checked for all the clusters and tries to set cluster
+    # status as `unhealthy` if the status field is missing (due to TTL)
+    for cluster in clusters.leaves:
+        int_id = cluster.key.split('/')[-1]
+        try:
+            if cluster.is_managed == "yes":
```
Comment (r0h4n) [inline tendrl/integrations/gluster/check_cluster_status.py]: I am not sure how you have tested this. etcd_utils.read returns EtcdResult. you cannot do a "cluster.is_managed" directly. Thoughts?

Comment (shtripat) [reply tendrl/integrations/gluster/check_cluster_status.py]: Updated the changes

**Pull request merged**